src/main/java/uk/gov/mca/beacons/service/registrations/RegistrationsService.java

```diff
@@ -1,27 +1,98 @@
 package uk.gov.mca.beacons.service.registrations;
 
+import java.util.List;
+import java.util.UUID;
+import javax.transaction.Transactional;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import uk.gov.mca.beacons.service.model.Beacon;
+import uk.gov.mca.beacons.service.model.BeaconPerson;
+import uk.gov.mca.beacons.service.model.BeaconStatus;
+import uk.gov.mca.beacons.service.model.BeaconUse;
+import uk.gov.mca.beacons.service.model.PersonType;
 import uk.gov.mca.beacons.service.model.Registration;
+import uk.gov.mca.beacons.service.repository.BeaconPersonRepository;
+import uk.gov.mca.beacons.service.repository.BeaconRepository;
+import uk.gov.mca.beacons.service.repository.BeaconUseRepository;
 
 @Service
+@Transactional
```

Comment:
Added `@Transactional` annotation on the service to ensure that if there are any unhandled 5xx errors thrown by the service - this is essentially a `RuntimeError` in Java, i.e: unhandled error by the service.

Then any changes to the database are automatically rolled back by Spring.  Under the hood Spring will manage the database transaction and revert changes if there are any errors.